src/lib/services/classifiersService.ts

```diff
@@ -7,12 +8,32 @@ export interface ClassifiersData {
     brands: string[];
     colors: string[];
     priceIntervals: string[];
+    productVariants: ProductVariant[];
 }
 
 export const getAllClassifiers = async (): Promise<ClassifiersData> => {
-    const { request } = classifiersService.getAll<ClassifiersData>();
-    const response = await request;
-    return Array.isArray(response.data) ? response.data[0] : response.data;
+    const { request: classifiersRequest } =
+        classifiersService.getAll<ClassifiersData>();
+    const { request: productsRequest } =
+        productService.getObject<APIProductData>();
+
+    const [classifiersResponse, productsResponse] = await Promise.all([
+        classifiersRequest,
+        productsRequest,
+    ]);
+
+    const classifiersData = Array.isArray(classifiersResponse.data)
+        ? classifiersResponse.data[0]
+        : classifiersResponse.data;
+
+    const productVariants = productsResponse.data.content.flatMap(
+        (product) => product.productVariants
+    );
+
+    return {
+        ...classifiersData,
+        productVariants,
+    };
```

Comment:
I am unsure whether this component should be using the productService. This results in a duplicate call to products (inspecting Network tab in dev tools):
![image](https://github.com/user-attachments/assets/4fc697e5-1c7d-45de-928b-d2f07aae3613)
Moreover classifier service does not include abort controller, which could be helpful in improving performance. I suggest to add an abort controller to this method.

Reply:
<img width="578" alt="Screenshot 2024-10-18 at 21 10 37" src="https://github.com/user-attachments/assets/263af138-33ce-478c-bf2d-3d10111838d6">